### PR TITLE
New topic: quantum resistance

### PR DESCRIPTION
**Pull request by harding**

### CODE REVIEW

File: _topics/en/quantum-resistance.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+- **ECDSA** public keys used in Bitcoin are vulnerable to a
+  factorization attack using [Shor's algorithm][].  This would
+  completely eliminate the security of ECDSA assuming an idealized
+  quantum computer.  Since public keys used for proposed [schnorr
+  signatures][topic schnorr signatures] are essentially identical to
+  those used for ECDSA, the same attack applies.  Quantum resistant
+  alternatives to ECDSA are known, but they involve much larger key and
+  signature sizes, so most developers seem to prefer to delay upgrading
+  until its necessary.
```
Comment (jonatack) [inline _topics/en/quantum-resistance.md:89]: it's

File: _topics/en/quantum-resistance.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  idealized quantum computer.  That means an algorithm like SHA256 that
+  currently has an estimated second preimage resistance of 256 bits
+  (2<sup>256</sup>) would be reduced to 128 bits (2<sup>128</sup> is the
+  square root of 2<sup>256</sup>).  That loss can be overcome by using a
+  roughly equivalent algorithm with twice as many security bits, e.g.
+  going from SHA256 to SHA512.
+
+- **ECDSA** public keys used in Bitcoin are vulnerable to a
+  factorization attack using [Shor's algorithm][].  This would
+  completely eliminate the security of ECDSA assuming an idealized
```
Comment (jonatack) [inline _topics/en/quantum-resistance.md:83]:
```suggestion
  completely eliminate the security of ECDSA, assuming an idealized
```

File: _topics/en/quantum-resistance.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  quantum computer.  Since public keys used for proposed [schnorr
+  signatures][topic schnorr signatures] are essentially identical to
+  those used for ECDSA, the same attack applies.  Quantum resistant
```
Comment (jonatack) [inline _topics/en/quantum-resistance.md:86]:
```suggestion
  those used for ECDSA, the same attack applies.  Quantum-resistant
```

File: _topics/en/quantum-resistance.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  implementation in other parts of Bitcoin, such as the [version 2
+  Bitcoin transport protocol][topic v2 p2p transport].
+
+The worst case for attacks assumes an idealized quantum computer with
+sufficient capacity and reliability to perform the attack.  It's likely
+that the capacity and reliability of quantum computers will increase
+gradually over time, meaning the security of the cryptography used in
+Bitcoin will similarly decrease gradually over time, with attacks
+progressing from computationally infeasible, to theoretically possible
+but implausible, to extraordinary expensive, to very expensive, to
```
Comment (jonatack) [inline _topics/en/quantum-resistance.md:106]:
```suggestion
but implausible, to extraordinarily expensive, to very expensive, to
```

File: _topics/en/quantum-resistance.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
+---
+title: Quantum resistance
+
+## Optional.  Shorter name to use for reference style links e.g., "foo"
+## will allow using the link [topic foo][].  Not case sensitive
+# shortname: foo
+
+## Optional.  An entry will be added to the topics index for each alias
+aliases:
+  - Post quantum cryptography
```
Comment (jonatack) [inline _topics/en/quantum-resistance.md:10]: like line 110
```suggestion
  - Post-quantum cryptography
```

**Review by jonatack — COMMENTED** [review]
Excellent addition! Thanks @harding.

File: _topics/en/quantum-resistance.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+#      link: https://example.com
+
+## Optional.  Each entry requires "title", "url", and "date".  May also use "feature:
+## true" to bold entry
+optech_mentions:
+  - title: Discussion of quantum computer attacks on taproot
+    url: /en/newsletters/2021/03/24/#discussion-of-quantum-computer-attacks-on-taproot
+    feature: true
+
+  - title: "Question about paying public keys directly versus hash indirection"
```
Comment (bitschmidty) [inline _topics/en/quantum-resistance.md:31]: The other questions were "Question:" prefixed so this one stood out to me.

File: _topics/en/quantum-resistance.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  factorization attack using [Shor's algorithm][].  This would
+  completely eliminate the security of ECDSA, assuming an idealized
+  quantum computer.  Since public keys used for proposed [schnorr
+  signatures][topic schnorr signatures] are essentially identical to
+  those used for ECDSA, the same attack applies.  Quantum-resistant
+  alternatives to ECDSA are known, but they involve much larger key and
+  signature sizes, so most developers seem to prefer to delay upgrading
+  until it's necessary.
+
+- **Noise** is the protocol used for encrypted communication in LN.
```
Comment (bitschmidty) [inline _topics/en/quantum-resistance.md:91]: I personal hadnt heard of Noise before and went out to google it. Perhaps others would benefit from a link?

https://noiseprotocol.org/ is the protocol website, but maybe an explanation like https://duo.com/labs/tech-notes/noise-protocol-framework-intro or another would be appropriate

**Review by bitschmidty — COMMENTED** [review]
Succinctly informative, thank you @harding !

**Comment by jnewbery** [comment]
LGTM. Let's merge once @harding has had a chance to respond to Mike's review comments.

File: _topics/en/quantum-resistance.md (hunk 6 above)
Comment (jnewbery) [reply _topics/en/quantum-resistance.md:31]: others have been changed to "Question about ..."

**Pull request merged**